NewHorizons/Utility/Files/ImageUtilities.cs

```diff
@@ -17,7 +19,72 @@ public static class ImageUtilities
         // key is path + applied effects
         private static readonly Dictionary<string, Texture> _textureCache = new();
         public static bool CheckCachedTexture(string key, out Texture existingTexture) => _textureCache.TryGetValue(key, out existingTexture);
-        public static void TrackCachedTexture(string key, Texture texture) => _textureCache.Add(key, texture);
+        public static void TrackCachedTexture(string key, Texture texture)
+        {
+            _textureCache[key] = texture;
+
+            // Save files to thing
+            if (texture is Texture2D texture2D)
+            {
+                try
+                {
+                    var fileName = $"{key.GetHashCode()}.png";
+
+                    var path = Path.Combine(Main.Instance.ModHelper.Manifest.ModFolderPath, "Cache", Main.Instance.CurrentStarSystem, fileName);
+                    Directory.CreateDirectory(Path.GetDirectoryName(path));
+                    File.WriteAllBytes(path, texture2D.EncodeToPNG());
```

Comment:
TODO in the future: use GetRawTextureData and LoadRawTextureData, which works when doing Texture2D.Compress. should be faster than encoding to png too since its just the raw byte array